base/range.jl

```diff
@@ -901,6 +901,8 @@ end
 
 ## indexing
 
+isassigned(r::AbstractRange, i::Int) = firstindex(r) <= i <= lastindex(r)
```

Comment:
Is this true for StepRange? I think it might only be true for Unit ranges

Reply:
I'm not aware of an instance where it could produce an undefined variable

Reply:
oh, right, I was thinking of values, not index